### PR TITLE
ENG-43804: updating provider for supporting multiple accounts config

### PR DESCRIPTION
**Pull request by varkey98**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,64 +1,72 @@
...
-	github.com/aws/aws-sdk-go-v2/credentials v1.12.8
-	github.com/aws/aws-sdk-go-v2/service/apigateway v1.16.11
-	github.com/aws/aws-sdk-go-v2/service/apigatewayv2 v1.13.11
-	github.com/aws/aws-sdk-go-v2/service/sts v1.16.9
-	github.com/golang/mock v1.2.0
-	github.com/google/uuid v1.3.0
-	github.com/hashicorp/terraform-plugin-docs v0.14.1
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.17.0
-	github.com/stretchr/testify v1.7.0
+	github.com/aws/aws-sdk-go-v2 v1.26.1
```
Comment (varkey98) [inline go.mod:6]: bumped every dependency to latest version, github is flagging vulnerabilities on the current main: https://github.com/Traceableai/terraform-provider-awsapigateway/security/dependabot

File: provider/resource.go [hunk 2]
```diff
@@ -23,52 +27,95 @@ func AwsApiGatewayResource() *schema.Resource {
...
 		diagnostics:      diag.Diagnostics{},
 		warnDiagnostics:  make(map[string][]string),
 		errorDiagnostics: make(map[string][]string),
 	}
-	logGroupNames := getLogGroupNames(apiGateways, action == string(EXCLUDE), ignoreAccessLogSettings, meta, mapDiagnostics)
+
+	conn := meta.(AwsApiGatewayProvider)
+
+	for _, account := range accounts {
+		acc := account.(map[string]interface{})
```
Comment (tim-mwangi) [inline provider/resource.go:87]: There's no chance that these type casts would error, right?

Comment (varkey98) [reply provider/resource.go:87]: No, if there's a type mismatch, then terraform will throw an error before calling into the provider as its strictly typed there

**Review by tim-mwangi — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -23,6 +23,45 @@ provider "awsapigateway" {
 
 See the complete example [here](./examples/default)
```
Comment (t-santoshsahu) [inline README.md:31]: @varkey98 Can you add some examples here, with new input's support

**Review by t-santoshsahu — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -11,18 +11,64 @@ terraform {
...
+resource "awsapigateway_resouce" "traceable-example-2" {
+  identifier                 = uuid()
+  ignore_access_log_settings = false
+  account {
+    region                 = "us-east-1"
+    api_list               = ["api1", "api2"]
+    cross_account_role_arn = ""
+    exclude                = false
+  }
 }
```
Comment (puneet-traceable) [inline README.md:28]: can add another one for include

Comment (varkey98) [reply README.md:28]: sure, sorry, just merged the pr :|

Comment (varkey98) [reply README.md:28]: @puneet-traceable https://github.com/Traceableai/terraform-provider-awsapigateway/pull/4

**Pull request merged**